runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -258,7 +260,8 @@ boolean atMaxWatermark(DataflowPipelineJob job, JobMetrics metrics) {
     for (MetricUpdate metric : metrics.getMetrics()) {
       if (metric.getName() == null
           || metric.getName().getName() == null
-          || !metric.getName().getName().endsWith(WATERMARK_METRIC_SUFFIX)
+          || !(metric.getName().getName().endsWith(LEGACY_WATERMARK_METRIC_SUFFIX)
```

Comment:
Can we extract this to a helper method or a variable. As is, reading the boolean expression is difficult to read. I think this could be:

```
private boolean isWatermark(MetricUpdate metric) {
  if (metric.getName() == null) || metric.getName().getName() == null) {
    return false; // no name -> shouldn't happen, not the watermark
  }
  if (metric.getScalar() == null) {
    return false; // no scalar value -> not the watermark
  }
  String name = metric.getName().getName();
  return name.endsWith(LEGACY_WATERMARK_METRIC_SUFFIX) 
      || name.endsWith(WATERMARK_METRIC_SUFFIX);
}
```